### PR TITLE
fix default value bug in inference and make FG type compatible with E…

### PR DESCRIPTION
**Pull request by dawn310826**
…asyRec input_type

### CODE REVIEW

File: easy_rec/python/inference/predictor.py [hunk 1]
```diff
@@ -135,6 +137,18 @@ def search_pb(self, directory):
 
     return dir_list[0]
 
+  def get_input_fields_from_pipeline_config(self, model_path):
+    pipeline_path = os.path.join(model_path, 'assets/pipeline.config')
+    assert tf.gfile.Exists(pipeline_path), '%s not exists.' % pipeline_path
+    pipeline_config = get_configs_from_pipeline_file(pipeline_path)
+    input_fields = pipeline_config.data_config.input_fields
+    input_fields_info = {
```
Comment (tiankongdeguiji) [inline easy_rec/python/inference/predictor.py:145]: input_fields_info -> effective_fields_info

File: easy_rec/python/inference/predictor.py [hunk 2]
```diff
@@ -363,7 +386,10 @@ def _get_defaults(col_type):
         tmp_cols.append(tmp_keys[0].strip())
       output_cols = tmp_cols
 
-    record_defaults = [_get_defaults(x) for x in all_col_types]
+    record_defaults = [
+        _get_defaults(col_name, col_type)
+        for col_name, col_type in zip(all_cols, all_col_types)
+    ]
```
Comment (tiankongdeguiji) [inline easy_rec/python/inference/predictor.py:392]: do not need select all cols.

**Review by lgqfhwy — APPROVED** [review]

File: easy_rec/python/utils/input_utils.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    return default_val.lower() == 'true'
+  elif field_type in [DatasetConfig.FLOAT]:
+    return float(default_val)
+  elif field_type in [DatasetConfig.DOUBLE]:
+    return np.float64(default_val)
+
+  return type_defaults[field_type]
+
+
+def string_to_number(record_default, cur_field, name=''):
```
Comment (tiankongdeguiji) [inline easy_rec/python/utils/input_utils.py:40]: make input params same as tf.string_to_number

Comment (tiankongdeguiji) [reply easy_rec/python/utils/input_utils.py:40]: add docstring

File: easy_rec/python/utils/input_utils.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  tmp_field = cur_field
+  if type(record_default) in [int, np.int32, np.int64]:
+    tmp_field = tf.string_to_number(
```
Comment (tiankongdeguiji) [inline easy_rec/python/utils/input_utils.py:43]: add comment `deal with scientific notation`

File: easy_rec/python/test/fg_test.py [hunk 5]
```diff
@@ -1,34 +1,57 @@
...
         pipeline_config, as_utf8=True)
 
     final_pipeline_config = config_util.get_configs_from_pipeline_file(
-        self._final_pipeline_config_path)
+        final_pipeline_config_path)
     final_pipeline_config_str = text_format.MessageToString(
         final_pipeline_config, as_utf8=True)
     self.assertEqual(pipeline_config_str, final_pipeline_config_str)
 
+  def test_fg_dtype(self):
```
Comment (tiankongdeguiji) [inline easy_rec/python/test/fg_test.py:50]: put the test into train_eval_test

**Review by tiankongdeguiji — CHANGES_REQUESTED** [review]

File: easy_rec/python/test/fg_test.py (hunk 5 above)
Comment (dawn310826) [reply easy_rec/python/test/fg_test.py]: done

File: easy_rec/python/utils/input_utils.py (hunk 4 above)
Comment (dawn310826) [reply easy_rec/python/utils/input_utils.py:53]: done

File: easy_rec/python/utils/input_utils.py (hunk 3 above)
Comment (dawn310826) [reply easy_rec/python/utils/input_utils.py]: done

File: easy_rec/python/inference/predictor.py (hunk 2 above)
Comment (dawn310826) [reply easy_rec/python/inference/predictor.py]: done

**Review by tiankongdeguiji — APPROVED** [review]
LGTM

**Pull request merged**